### PR TITLE
K8SPXC-510 Use ENV variables for images

### PR DESCRIPTION
**Pull request by cap1984**
[K8SPXC-510](https://jira.percona.com/browse/K8SPXC-510)

### CODE REVIEW

File: pkg/apis/pxc/v1/pxc_types.go [hunk 1]
```diff
@@ -432,6 +441,10 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *version.ServerV
 		if len(c.HAProxy.ImagePullPolicy) == 0 {
 			c.HAProxy.ImagePullPolicy = corev1.PullAlways
 		}
+		if len(os.Getenv("RELATED_IMAGE_HAPROXY")) != 0 {
+			c.HAProxy.Image = os.Getenv("RELATED_IMAGE_HAPROXY")
+			c.HAProxy.ImagePullPolicy = corev1.PullIfNotPresent
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go:446]: why we need to rewrite already set policy?

File: pkg/apis/pxc/v1/pxc_types.go [hunk 2]
```diff
@@ -360,6 +361,10 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *version.ServerV
 		if len(c.PXC.ImagePullPolicy) == 0 {
 			c.PXC.ImagePullPolicy = corev1.PullAlways
 		}
+		if len(os.Getenv("RELATED_IMAGE_PXC")) != 0 {
+			c.PXC.Image = os.Getenv("RELATED_IMAGE_PXC")
+			c.PXC.ImagePullPolicy = corev1.PullIfNotPresent
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go:366]: policy can be set via CR, I believe we don't need to change it here

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| recreate | passed |
| haproxy | passed |
| smart-update | failed |
| scaling | passed |
| big-data | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| smart-update | failed |
| scaling | passed |
| haproxy | passed |
| storage | failed |
| recreate | passed |
| big-data | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| self-healing | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| users | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | failed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| smart-update | failed |
| storage | failed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| limits | passed |
| upgrade | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| security-context | passed |
| big-data | passed |
| storage | failed |
| smart-update | failed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| affinity | failed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| smart-update | failed |
| haproxy | passed |
| storage | passed |
| recreate | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| operator-self-healing | failed |
| affinity | passed |
| demand-backup | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |
| haproxy | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| big-data | passed |
| operator-self-healing | failed |
| monitoring-2-0 | passed |
| scaling | passed |
| affinity | passed |
| scaling-proxysql | passed |
| one-pod | passed |
| upgrade | failed |
| smart-update | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| smart-update | failed |
| recreate | passed |
| scaling | passed |
| storage | passed |
| haproxy | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| operator-self-healing | failed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**